utils/src/singleflight.rs

```diff
@@ -42,30 +42,31 @@ use std::future::Future;
 use std::marker::PhantomData;
 use std::pin::Pin;
 use std::sync::atomic::{AtomicBool, AtomicU16, Ordering};
-use std::sync::{Arc, RwLock};
+use std::sync::{Arc, Mutex, RwLock};
 use std::task::{ready, Context, Poll};
 
 use futures::future::Either;
 use pin_project::{pin_project, pinned_drop};
 use tokio::runtime::Handle;
-use tokio::sync::{Mutex, Notify};
```

Comment:
Note: this PR switches from a Tokio Mutex to a std Mutex since we can't call async code in `drop()` (and we need to lock the `CallMap` to remove the Call on `CreateGuard::drop`). 

This is safe to do since the locking sections are only over data (i.e. the `CallMap`) and not across `await` boundaries or I/O.